action.yaml

```diff
@@ -145,46 +145,55 @@ runs:
         error_end_message="Go to the Code Scanning section of the GitHub Security tab to review these vulnerabilities."
         error_search_pr_message="Search for is:open pr:"$pr_number" to find PR related vulnerabilities."
 
-        if [[ $is_high_vuln_present == 'false' &&  $is_critical_vuln_present == 'false' ]]
-        then
-          echo "Success! No high or critical code scanning alerts.";
-          exit 0;
-        fi
+        _message="Undefined. Bug in this action. Fix."
+        _exit_code=0
 
-        if [[ $ALLOW_SEVERITY_LEVEL == 'medium' ]]
+        if [[ $is_high_vuln_present == 'false' &&  $is_critical_vuln_present == 'false' ]]
         then
-          if [[ ${{ github.event_name }} == 'pull_request' ]]
+          _message="Success! No high or critical code scanning alerts."
+          _exit_code=0;
+        else
+          if [[ $ALLOW_SEVERITY_LEVEL == 'medium' ]]
           then
-            echo ""$error_start_message" high or critical. "$error_end_message" "$error_search_pr_message"";
-            exit 1;
-          else
-            echo ""$error_start_message" high or critical found on branch "$REF_NAME". $error_end_message";
-            exit 1;
-          fi
-
-        elif [[ $ALLOW_SEVERITY_LEVEL == 'high' ]]
-        then
-          if [[ $is_critical_vuln_present == 'false' ]]
+            if [[ ${{ github.event_name }} == 'pull_request' ]]
+            then
+              _message="$error_start_message high or critical. $error_end_message $error_search_pr_message";
+              _exit_code=1
+            else
+              _message="$error_start_message high or critical found on branch '$REF_NAME'. $error_end_message";
+              _exit_code=1;
+            fi
+  
+          elif [[ $ALLOW_SEVERITY_LEVEL == 'high' ]]
           then
-            echo "Only high vulnerabilities detected! Allowing due to input ALLOW_SEVERITY_LEVEL being set to high.";
-            exit 0;
-          fi
-
-          if [[ ${{ github.event_name }} == 'pull_request' ]]
+            if [[ $is_critical_vuln_present == 'false' ]]
+            then
+              _message="Only high vulnerabilities detected! Allowing due to input ALLOW_SEVERITY_LEVEL being set to high.";
+              _exit_code=0;
+            fi
+  
+            if [[ ${{ github.event_name }} == 'pull_request' ]]
```

Comment:
Tidligere algoritme var basert på fail fast, mens omskrevet flyt feiler til slutt via statefull parameter `_exit_code`.

Flyten blir her endret slik at man alltid får exit code 1, uansett for caset `ALLOW_SEVERITY_LEVEL == 'high'`.
Dette ser ut til å produsere  feilen som nevnt på https://kartverketgroup.slack.com/archives/C04FFERQWNQ/p1748853675052389

Jeg antar at de er en `elif` som var tenkt her?